pkg/sql/pgwire/conn.go

```diff
@@ -562,12 +563,15 @@ func (c *conn) processCommandsAsync(
 			if !authOK {
 				ac.AuthFail(retErr)
 			}
+			if authCleanup != nil {
```

Comment:
I don't understand why this is called so late -- this call will only occur at the end when the client closes the connection. Any way to call that just after the authn completes?

Reply:
It specifically needs to be called on connection close, not just post-auth. See https://github.com/postgres/postgres/blob/f4d59369d2ddf0ad7850112752ec42fd115825d4/src/backend/libpq/pqcomm.c#L269. Also see the discussion at https://tools.ietf.org/html/rfc2744.html under section `1. Introduction` listing the four stages of using the GSS-API, stage `d`:

> At the completion of a communications session (which may extend
>      across several transport connections), each application calls a
>      GSS-API routine to delete the security context.  Multiple contexts
>      may also be used (either successively or simultaneously) within a
>      single communications association, at the option of the
>      applications.

I've added a comment to detail this.

Reply:
Ok but then the word "cleanup" is not the right word.
We have precedent throughout the crdb code base that "if a cleanup fn is returned, the caller should defer its call immediately"

I think here the proper name is "connCloseAuthHandler" or similar, to spell out in the name where the call should happen.

(I can even imagine a situation down the line where we have *both* an immediate "cleanup" function and a "conn close handler" side-by-side)

Reply:
Renamed.